src/main/java/de/fzi/ros_as_a_service/impl/ValueNode.java

```diff
@@ -75,27 +75,21 @@ public ValueInputNode(String label, String type, String value, boolean is_array)
   }
 
   static public ValueType getTypeFromString(String type_str) {
-    switch (type_str) {
-      case "string":
-        return ValueType.STRING;
-      case "int8":
-      case "int16":
-      case "int32":
-      case "int64":
-        return ValueType.INTEGER;
-      case "uint8":
-      case "uint16":
-      case "uint32":
-      case "uint64":
-        return ValueType.UINTEGER;
-      case "float32":
-      case "float64":
-        return ValueType.FLOAT;
-      case "bool":
-        return ValueType.UINTEGER;
-      default:
-        return ValueType.UNKNOWN;
+    if (type_str.equals("string")) {
+      return ValueType.STRING;
+    } else if (type_str.equals("int8") | type_str.equals("int16") | type_str.equals("int32")
+        | type_str.equals("int64")) {
```

Comment:
Should `|` be `||`? IIRC, `|` is a bitwise-or, not a boolean-or.

Reply:
From my understanding, java is a bit different here: https://stackoverflow.com/a/1808879

But again, I'm no java expert.

Reply:
`||` is the short-circuiting variant. If you did not want short-circuiting, then `|` would be OK.

From the expression, it doesn't seem like it would be needed to evaluate of all those `equals(..)`: the first match would make the category clear I believe.

Reply:
You mean if we matched against `"int*"`?